### DISCUSSION TITLE
M2strm Qnap nas

### DISCUSSION THREAD

**Discussion by Stefan765** [trix7777/m2strm#8] [General]
Hello!
I have a qnap nas and wan‘t to run your m2strm script on it for emby. I think it has to work with the mono app, but is it possible that the script automatically runs once a day or so…?
Thank you!
Stefan

**Comment by laurent734** [comment]
you need to configure cron tab on your nas.
m2strm works great with mono.

**Comment by Stefan765** [comment]
Thanks for your help!
I will try, maybe i need your help again, because i am not a pro on linux😉

**Comment by laurent734** [comment]
you can also use it on windows and define an smb url to a nas file.
in the m2strm.exe.config configuration file:

**Comment by Stefan765** [comment]
I need your help again! I have installed the mono app in Qts, but i don't know how to start it. Can you give me a step by step guide ....?

When i use it on my windows laptop, i have to run the laptop all time, or i am wrong?!

**Comment by laurent734** [comment]
ok to guide you but i'm not the m2strm developer.
did you manage to get it to work on windows?

**Comment by laurent734** [comment]
if you run it on windows, you can choose a network path to generate strm files directly in your nas.
in the m2strm.exe.config file:
<add key="OutDirectory" value="\\COREELEC\TV Shows\" />
add key = "outDirectory" value="your url smb nas"\"/

**Comment by Stefan765** [comment]
thank you! in windows it works, but when i add as outdirectory my nas i need my laptop to download the m3u.

**Comment by laurent734** [comment]
for mono you must connect in ssh on your nas.
install putty on your windows pc for this.
connect to putty with the nas ip and the nas admin password.
once connected the command are almost identical, you just have to add mono in front.
ex on windows

C: \ Users \ laure \ Documents \ m2strm / m2strm.exe / m

on linux with mono

sudo mono ./your folder path / m2strm.exe / m
 
I don't know the folder structure of your nas so you have to adapt it yourself.

**Comment by trix7777** [comment]
After getting m2strm to run manually to make sure it works as expected and after you've configured it, edit crontab by:
`crontab -e`
(you might get the question to use nano or vi, I suggest you chose nano)
add this (example):
`5 7-23/2 * * * /path/to/mono /path/to/m2strm.exe`
and save.
This will run the app "At minute 5 past every 2nd hour from 7 through 23"

**Comment by trix7777** [comment]
Btw, this is not an issue or a bug. Ask further questions in Discussions.
